lib/modules/platform/gerrit/index.ts

```diff
@@ -0,0 +1,553 @@
+      changes.length;
+    if (allSubmittable) {
+      return 'green';
+    }
+    const hasProblems =
+      changes.filter((change) => change.problems.length > 0).length > 0;
+    if (hasProblems) {
+      return 'red';
+    }
+  }
+  return 'yellow';
+}
+
+/**
+ * check the gerrit-change for the presence of the corresponding "$context" Gerrit label if configured,
+ *  return 'yellow' if not configured or not set
+ * @param branchName
+ * @param context renovate/stability-days || ...
+ */
+export async function getBranchStatusCheck(
+  branchName: string,
+  context: string | null | undefined
+): Promise<BranchStatus | null> {
+  const { labelName } = mapBranchStateContextToLabel(context);
+  if (labelName) {
+    const change = (await findOwnPr({ branchName, state: 'open' }, true)).pop();
+    if (change) {
+      const labelRes = change.labels?.[labelName];
+      if (labelRes) {
+        if (labelRes.approved) {
+          return 'green';
+        }
+        if (labelRes.rejected) {
+          return 'red';
+        }
+      }
+    }
+  }
+  return 'yellow';
+}
+
+/**
+ * Apply the branch state $context to the corresponding gerrit label (if available)
+ * context === "renovate/stability-days" / "renovate/merge-confidence" and state === "green"/...
+ * @param branchStatusConfig
+ */
+export async function setBranchStatus(
+  branchStatusConfig: BranchStatusConfig
+): Promise<void> {
+  const { labelName, label } = mapBranchStateContextToLabel(
+    branchStatusConfig.context
+  );
+  const labelValue =
+    label && mapBranchStatusToLabel(branchStatusConfig.state, label);
+  if (labelName && labelValue) {
+    const pr = await getBranchPr(branchStatusConfig.branchName);
+    if (pr === null) {
+      return Promise.resolve();
+    }
+    await client.setLabel(pr.number, labelName, labelValue);
+  }
+  return Promise.resolve();
+}
+
+function mapBranchStateContextToLabel(context: string | null | undefined): {
+  labelName?: string;
+  label?: GerritLabelTypeInfo;
+} {
+  let labelName;
+  switch (context) {
+    case 'renovate/stability-days':
+      labelName = config.labelMappings?.stabilityDaysLabel;
+      break;
+    case 'renovate/merge-confidence':
+      labelName = config.labelMappings?.mergeConfidenceLabel;
+      break;
+  }
+  if (labelName && config.labels[labelName]) {
+    return {
+      labelName,
+      label: config.labels[labelName],
+    };
+  }
+  return {};
+}
+
+export function getRawFile(
+  fileName: string,
+  repoName?: string,
+  branchOrTag?: string
+): Promise<string | null> {
+  const repo = repoName ?? config.repository ?? 'All-Projects';
+  const branch = branchOrTag ?? config.head ?? 'HEAD';
+  return client.getFile(repo, branch, fileName);
+}
+
+export async function getJsonFile(
+  fileName: string,
+  repoName?: string,
+  branchOrTag?: string
+): Promise<any | null> {
+  const raw = (await getRawFile(fileName, repoName, branchOrTag)) as string;
+  return JSON5.parse(raw);
```

Comment:
add a proper check instead of the cast. other platforms need to be changed too later